library/src/main/java/kwasm/format/text/ParseResult.kt

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kwasm.format.text
+
+import kwasm.ast.AstNode
+
+data class ParseResult(val astNode: AstNode, val nextIndex: Int)
```

Comment:
It might be cleaner to make this generic.
```suggestion
data class ParseResult<T : AstNode>(val astNode: T, val nextIndex: Int)
```